utilities/transactions/transaction_lock_mgr.cc

```diff
@@ -97,6 +97,47 @@ struct LockMap {
   size_t GetStripe(const std::string& key) const;
 };
 
+void DeadlockInfoBuffer::AddNewPath(std::vector<DeadlockInfo>& path) {
+  std::lock_guard<std::mutex> lock(paths_buffer_mutex_);
+  buffer_idx = (buffer_idx + 1) % paths_buffer.size();
+  paths_buffer[buffer_idx] = path;
+}
+
+std::vector<std::vector<DeadlockInfo>> DeadlockInfoBuffer::prepareBuffer() {
+  std::lock_guard<std::mutex> lock(paths_buffer_mutex_);
+  std::vector<std::vector<DeadlockInfo>> working;
+  for (size_t i = buffer_idx, counter = 0; counter < paths_buffer.size();
+       counter++) {
+    if (!paths_buffer[i].size()) {
+      break;
+    }
+
+    working.push_back(paths_buffer[i]);
+    if (i == 0) {
+      i = paths_buffer.size();
+    }
+    i--;
+  }
+  return working;
+}
+
+DeadlockInfoBuffer::DeadlockInfoBuffer(uint32_t n_latest_dlocks) {
+  paths_buffer = std::vector<std::vector<DeadlockInfo>>(n_latest_dlocks);
+}
+
+void DeadlockInfoBuffer::DynamicResize(uint32_t target_size) {
```

Comment:
Yes. The size of the buffer should be dynamically configurable. I'm going to add max_deadlock_count as a TransactionDBOption to have the size of the buffer configurable.